File: backend/src/main/java/com/travelcompanion/entity/Trip.java
package com.travelcompanion.entity;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDate;

@Entity
@Table(name = "trips")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Trip {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "destination_id")
    private Destination destination;

    @Column(nullable = false)
    private LocalDate startDate;

    @Column(nullable = false)
    private LocalDate endDate;

    private Double budget;

    @Enumerated(EnumType.STRING)
    private TravelType travelType;

    private String notes;

    @Enumerated(EnumType.STRING)
    private TripStatus status = TripStatus.PLANNED;

    public enum TravelType {
        SOLO, COUPLE, FAMILY, ADVENTURE, GROUP, BUSINESS
    }

    public enum TripStatus {
        PLANNED, ONGOING, COMPLETED, CANCELLED
    }
}

File: backend/src/models/Trip.js
const mongoose = require('mongoose');

const tripSchema = new mongoose.Schema(
  {
    userId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'User',
      required: [true, 'User ID is required'],
      index: true,
    },
    destinationId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'Destination',
    },
    startDate: {
      type: Date,
      required: [true, 'Start date is required'],
    },
    endDate: {
      type: Date,
      required: [true, 'End date is required'],
    },
    budget: {
      type: Number,
      default: 0,
    },
    travelType: {
      type: String,
      enum: ['SOLO', 'COUPLE', 'FAMILY', 'ADVENTURE', 'GROUP', 'BUSINESS'],
      default: 'SOLO',
    },
    notes: {
      type: String,
      default: '',
    },
    status: {
      type: String,
      enum: ['PLANNED', 'ONGOING', 'COMPLETED', 'CANCELLED'],
      default: 'PLANNED',
    },
  },
  {
    timestamps: true,
    toJSON: {
      virtuals: true,
      transform: (doc, ret) => {
        ret.id = ret._id;
        delete ret.__v;
        return ret;
      },
    },
    toObject: {
      virtuals: true,
      transform: (doc, ret) => {
        ret.id = ret._id;
        delete ret.__v;
        return ret;
      },
    },
  }
);

// Virtual id field
tripSchema.virtual('id').get(function () {
  return this._id.toHexString();
});

// Indexes
tripSchema.index({ userId: 1, status: 1 });
tripSchema.index({ startDate: 1, endDate: 1 });

module.exports = mongoose.model('Trip', tripSchema);
